### DISCUSSION TITLE
[Feature idea] Have a way to prevent `yarn cache clean`

### DISCUSSION THREAD

**Discussion by jakub-g** [yarnpkg/berry#6518] [Ideas]
# Context

_(For context: we have a huge repo with 1000s of deps, use pnp mode, and commit `.yarn/cache` to git)._

For some reason, some folks in my company, whenever having an issue, try `yarn cache clean` as a general troubleshooting option (perhaps they're used to `rm -rf node_modules` from old times of npm; also, they might be conflating `yarn cache clean` with some other commands).

This creates more problems than it solves:

- it creates a massive `git diff` on `.yarn/cache`
- things generally get more broken than before, and running `yarn install` is not fixing it.
  - It takes a lot of time to fetch all deps, run scripts, and eventually it fails.
  - This is e.g. because some of the deps deep in the dependency chain have `script`s which no longer work etc. so a fresh install from scratch with zero disk cache does not work anymore.
    - (This is a separate thing that perhaps should be fixed on our side internally... But as you can imagine, fixing outdated deps-of-deps-of-deps is tricky)
- so the folks don't know what to do, and come to ask for support
- the proper way to fix things it to undo the changes in `.yarn/cache` first, and from there, try to fix the original problem they had.

The bottom line is, _`yarn cache clean` is never a solution to the problems folks have, and it actually makes things worse_.

# Potential solution

- perhaps `yarn cache clean` (especially in pnp mode) should show a confirmation prompt? (or require `--force`)
  - might be a semver-major for yarn5
- having possibility to disable it (or require confirmation) via a yarnrc entry could also be an option.

**Comment by jakub-g** [comment]
Addressed in https://github.com/yarnpkg/berry/pull/6610
